util/chunk/disk.go

```diff
@@ -85,7 +87,7 @@ func NewListInDisk(fieldTypes []*types.FieldType) *ListInDisk {
 }
 
 func (l *ListInDisk) initDiskFile() (err error) {
-	l.disk, err = ioutil.TempFile(tmpDir, l.diskTracker.Label().String())
+	l.disk, err = ioutil.TempFile(initTempDir(), l.diskTracker.Label().String())
```

Comment:
Each new instance of `ListInDisk` will call `initDiskFile`, while `initTempDir()` removes the directory each time when it's called. So a `ListInDisk` may remove the files which are obtained by another `ListInDisk`s.